safe/impact_function_v4/impact_function.py

```diff
@@ -644,7 +652,10 @@ def prepare(self):
                 )
                 return PREPARE_FAILED_BAD_INPUT, message
 
+            self._provenance['exposure_layer'] = self.exposure.source()
```

Comment:
Hum, I'm confused. If you call self.exposure.source() in this function, you will get the source from the last exposure layer, not the original one. `performance_log` is called at the end. Same for hazard and aggregation